doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+| kuksa.val.v1 <sup>(3)</sup> | sync  |   ~  9 000 / s |
+| databroker   <sup>(4)</sup> | sync  |   ~ 26 000 / s |
+
+<sup>1</sup> kuksa_client is using `kuksa_client.KuksaClientThread()`
+
+<sup>2</sup> kuksa.grpc is using `kuksa_client.grpc` without the legacy `kuksa_client.KuksaClientThread()` wrapping it
+
+<sup>3</sup> uses the generated `val_pb2*.py` python lib directly
+
+<sup>4</sup> uses the generated `collector_pb2*.py` python lib directly
+
+
+Improvements:
+* Higher throughput.
+* Reduced CPU load.
+* Lower latency (probably, hasn't been measured)
+
+What's needed:
+* Introduce a streaming (GRPC) interface for providing sensor data.
+
+## Throttling-mode when system is overloaded.
+Is it worth to consider some throttling mode to be activated by the user in case system or any component is overloaded?
+Throttling modes to think about:
+  * Rate Limiting
+  * Bandwidth Throttling
+  * CPU Throttling
+
+## Considerations regarding shared and zero-copy memory approaches
+Pros:
+Cons:
+
+## Provider control and provider capabilities
+
+Open questions:
+
+Should the "change type" of a sensor (i.e. CONTINUOUS vs ON_CHANGE) be decided by providers
+or in the VSS metadata?
+It only makes sense for consumers to request their preferred rate of updates when they are
+subscribing to a sensor of type CONTINUOUS. That would be an argument for providing this
+information as part of the VSS metadata, so that it doesn't vary between vehicles.
+
+## Control the rate of updates
+
+Users of (continuous) sensor data can have different preferences with regard to how often
+they would like to receive updates. E.g. Vehicle.Speed is updated 100 times per second, but
+a consumer would only need it 1 time per second. This would introduce unnecessary processing
+requirements on the consumer (and provider).
+
+Currently there is no way for databroker to know how often a provider should provide updates.
+There is also no way for clients to instruct databroker of what rate they want.
+Sensor data is just sent at the rate it is received and providers are just sending sensor data at the rate
+they themselves decide.
+
+If a consumer can communicate this wish, there are several options for improvement.
+
+Improvements:
+* Reduction in load for consumers by adapting the update rate based on their preferences.
+* Reduction in load for for providers by propagating needed update rate.
+* Reduction in load for databroker by disabling unneeded providers.
+
+What's needed:
+* Introduce a way for clients to tell databroker of their preferred rate of updates.
+* Introduce a way for databroker to tell providers of highest needed frequency of sensor
+  data to which they can then adapt.
+  [probably needs] open stream `databroker` -> `provider`
+
+Other considerations:
+
+Setting the desired rate of update would only make sense for sensors of type CONTINUOUS.
+Sensors of type ON_CHANGE would always provide updates when the value changes.
```

Comment:
An interesting longterm topic for ON_CHANGE could be to discuss if we need a possibility to specify a threshold for what is considered as a changed value. Taking tire pressure as example, most likely every reading will be slightly different, but if value change from 2.305 to 2.307 it does not really matter. Setting a threshold of for example 0.2 could be useful if you want to know about significant changes as fast as possible.